calicoctl/commands/ipam/configure.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2016,2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	docopt "github.com/docopt/docopt-go"
+
+	"github.com/projectcalico/calicoctl/calicoctl/commands/clientmgr"
+	"github.com/projectcalico/calicoctl/calicoctl/commands/constants"
+	"github.com/projectcalico/libcalico-go/lib/ipam"
+)
+
+func updateIPAMStrictAffinity(ctx context.Context, ipamClient ipam.Interface, enabled bool) error {
+	ipamConfig, err := ipamClient.GetIPAMConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("Error: %v", err)
+	}
+
+	// If StrictAffinity == true => an address from a block can only be assigned by
+	// host with block affinity.
+	ipamConfig.StrictAffinity = enabled
+
+	err = ipamClient.SetIPAMConfig(ctx, *ipamConfig)
+	if err != nil {
+		return fmt.Errorf("Error: %v", err)
+	}
+
+	fmt.Println("Successfully set StrictAffinity to:", enabled)
+
+	return nil
+}
+
+// Configure IPAM.
+func Configure(args []string) error {
+	doc := constants.DatastoreIntro + `Usage:
+  calicoctl ipam configure --strictaffinity=<true/false> [--config=<CONFIG>]
+
+Options:
+  -h --help                        Show this screen.
+     --strictaffinity=<true/false> Set StrictAffinity to true/false. When StrictAffinity
+                                   is true, borrowing IP addresses is not allowed.
+  -c --config=<CONFIG>             Path to the file containing connection configuration in
+                                   YAML or JSON format.
+                                   [default: ` + constants.DefaultConfigPath + `]
+
+Description:
+  When a new Pod is scheduled on a node, IPAM assigns an IP address to that Pod
```

Comment:
I think this docstring probably doesn't make sense any more, for a few reasons:

- IPAM is used for more than just pods
- The configure subcommand is more generic than just IP borrowing (in theory). 

It could probably be as simple as something like "Modify and view configuration for Calico IP address management" 

Which makes me think, do we have a way for the user to view what the current state is?

Maybe we need a `calicoctl ipam show configuration` subcommand?